service/db.hpp

```diff
@@ -6,52 +6,79 @@
+                        config_key  text,
+	                readonly_config_key text
+                      );
+                    )"};
+  inline constexpr const db_statement_st create_unique_index_config_id_statement{
+      R"(create unique index if not exists config_id_uindex on config (id);)"};
+  inline constexpr const db_statement_st create_unique_index_config_key_statement{
+      R"(create unique index if not exists config_config_key_uindex on config (config_key);)"};
+  inline constexpr const db_statement_st create_unique_index_readonly_config_key_statement{
+      R"(create unique index if not exists config_readonly_config_key_uindex on config (readonly_config_key);)"};
+
+  struct config_create_answer_db
+  {
+    config_key_st config_key;
+    config_key_st readonly_config_key;
+    config_id_st config_id;
+  };
+
   class config_db
   {
   public:
     config_db(std::filesystem::path path_to_db) noexcept : database_{path_to_db.string()}
     {
         try {
-            database_ << R"(create table if not exists config
-                      (
-                        config_text text,
-                        id          integer not null
-                        constraint  table_name_pk
-                        primary key autoincrement
-                      );
-                    )";
-
-            database_ << R"(
-                            create unique index if not exists table_name_id_uindex on config (id);
-                        )";
+            database_ << create_table_statement.value();
+            database_ << create_unique_index_config_id_statement.value();
+            database_ << create_unique_index_config_key_statement.value();
+            database_ << create_unique_index_readonly_config_key_statement.value();
         }
         catch (const std::exception &error) {
             std::cerr << "error db occured: " << error.what() << std::endl;
         }
     }
 
-    using config_id = st::type<int, struct config_id_tag, st::arithmetic>;
-
-    config_id config_create(const json::json &config_create_data) noexcept
+    config_create_answer_db config_create(const json::json &config_create_data) noexcept
     {
         using namespace std::string_literals;
+        config_key_st config_key;
+        config_key_st readonly_config_key;
         try {
             json::json data_to_bind;
             data_to_bind["CONFIG_NAME"] = config_create_data["CONFIG_NAME"];
             data_to_bind["SETTINGS"] = {};
             std::cout << data_to_bind.dump() << std::endl;
             auto data_to_bind_str = data_to_bind.dump();
-            database_ << "insert into config (config_text) VALUES (?);" << data_to_bind_str;
+            database_ << "insert into config (config_text, config_key, readonly_config_key) VALUES (?, ?, ?);"
+                      << data_to_bind_str
+                      << (random_string() + std::to_string(std::hash<std::string>()(config_create_data["CONFIG_NAME"])))
```

Comment:
Even if you use hashes, you could have hash collisions. You need to handle the potential key collision

Reply:
I'm not fan of using an extra container for this here. if there is any collision an exception will occur and this will be reported to the client that can retry to send a create command if he want.

From the client point of view there is no extra cost to re-send the command, from the service we need an extra container (more memory etc...)

Reply:
I would just check if the db complains that the value isn't unique. No need for extra containers.